src/kademlia/routing_table.cpp

```diff
@@ -1198,4 +1184,18 @@ bool routing_table::is_full(int const bucket) const
 		&& int(i->replacements.size()) >= m_bucket_size);
 }
 
+#ifndef TORRENT_DISABLE_LOGGING
+void routing_table::log_node_failed(node_id const& nid, bucket_t::iterator const j) const
```

Comment:
I think it would be better to pass in a ``node_entry const&`` instead of an iterator here